### DISCUSSION TITLE
Cannot discover devices with custom prefixes

### DISCUSSION THREAD

**Discussion by dbahles** [jziolkowski/tdm#86] [Ideas]
I have change the prefix 1 & 2 for  camd and stat to a custom message "totas & "fromtas" I can no longer get the auto discover to find the devices.

I tried using the full message "topic and prefix" "sonoff/totas/"  but now the program crashes and will not start.

**Comment by jziolkowski** [comment]
custom prefixes are not yet supported.

**Comment by dbahles** [comment]
OK thank you - do you have a time frame?

**Comment by jziolkowski** [comment]
It is on the roadplan, but due to time constraints and RL issues I can't tell you any date. But you're not the first one to ask for this feature, so saty tuned.

**Comment by dbahles** [comment]
Thank you Jacek for the update

I really like your dashboard please keep up the great work

Dave

**Comment by ultima-originem** [comment]
Any update on this feature request? Unfortunately the lack of this feature prevents me from using TDM. A simple manual edit of the .cfg file would be OK for me.

**Reply by Flecky13** [reply]
Is there any news. it would be great.

**Comment by MarkPoticha** [comment]
Hi @jziolkowski,

TDM is a greate tool, thank You!

I have also the problem to get discoverd all the Tasmota MQTT devices with TDM at home.
my topics are like "shome/part_of_home/%topic%/%prefix%/"
In tdm.cfg you have the list of Patterns, you could generarte the list with the following script:
python.exe .\tasmota-mqtt-scan.py -h host --user <user> --password <password>
it works for me, if it works fou You too, may by You'd like to integrate it to TDM

CU

Mark

```
import paho.mqtt.client as mqtt 
import time
import json
import sys, getopt

Connected = False 
my_count = 0
def on_message(client, userdata, message):
    t = message.topic
    ts = t.split("/")
    if len(ts)>0:
        if ts[len(ts)-1] == "config":
            #print("device id:", ts[len(ts)-2])
            j = json.loads(message.payload.decode("utf-8"))
            global my_count 
            print(my_count,"=",j['ft'])
            my_count+=1            
   
  
def on_connect(client, userdata, flags, rc):
  
    if rc == 0:
        print("Connected to broker")
        global Connected                #Use global variable
        Connected = True                #Signal connection 
    else:
        print("Connection failed")

def read_config(broker_address,username,password):
    print("creating new instance")
    client = mqtt.Client("tasmota-mqtt-reader",) #create new instance
    if username !='':
        client.username_pw_set(username,password)
    client.on_connect=on_connect #attach function to callback
    client.on_message=on_message #attach function to callback

    print("connecting to broker")
    client.connect(broker_address) #connect to broker
    client.loop_start() #start the loop

    while Connected != True:    #Wait for connection
        time.sleep(0.1)
        
    print("Subscribing to topic","tasmota/discovery/#")
    client.subscribe("tasmota/discovery/#")

    time.sleep(10)
    
    print ("exiting")
    client.disconnect()
    client.loop_stop()


def main(argv):
    hostname = ''
    username = ''
    password = ''
    try:
        opts, args = getopt.getopt(argv,"h:up",["host=","user=","password="])
    except getopt.GetoptError:
        print ('tasmota-mqtt.py -h <hostname> -u <username> -p <password>')
        sys.exit(2)
    for opt, arg in opts:
        if opt == '-h':
            hostname = arg         
        elif opt in ("-u", "--user"):
            username = arg
        elif opt in ("-p", "--password"):
            password = arg
    print ('Host', hostname)
    print ('user', username)
    print ('password', password)
    read_config(hostname,username,password)
    
if __name__ == "__main__":
   main(sys.argv[1:])
```

**Reply by skupu** [reply]
@jziolkowski Any news?

**Reply by barbudor** [reply]
This is supported
Have you entered your prefix in the "MQTT -> Autodiscovery pattern" configuration window ?

**Reply by skupu** [reply]
I did. But maybe I made a mistake? The base_topic is "sensor/pzem_energy"

and then:
        String voltageTopic = String(base_topic) + "/voltage_" + String(i);
        String currentTopic = String(base_topic) + "/current_" + String(i);
        String powerTopic = String(base_topic) + "/power_" + String(i);
        String energyTopic = String(base_topic) + "/energy_" + String(i);
        String frequencyTopic = String(base_topic) + "/frequency_" + String(i);
        String pfTopic = String(base_topic) + "/power_factor_" + String(i);

This is arduino device.
It is visible in mqtt in HA.

**Reply by jziolkowski** [reply]
You are aware that TDM is only for Tasmota firmware?

**Reply by skupu** [reply]
yes, just wanted to catch if device is visible, which would be an advantage.

**Reply by jziolkowski** [reply]
To have the device visible in tdm during auto discovery, the following must happen:
- device has retained LWT topic and message
- device must have a fulltopic pattern
- device must respond to command template and few status commands. All these responses must be using correct tasmota payloads. 

Do you want to code all this only to have a faux device visible in TDM?
